### PR TITLE
Update/tests

### PR DESCRIPTION
**Pull request by kspeer825**
# Description of change
https://stitchdata.atlassian.net/browse/SQA-149

# Manual QA steps
 - Run tests locally and on branch in sandbox prior to merge
 
# Risks
 - corrupting test account
 
# Rollback steps
 - revert this branch
 - reset test account if necessary

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -11,7 +11,7 @@
     py_modules=["tap_stripe"],
     install_requires=[
         "singer-python==5.5.1",
-        "stripe==2.10.1",
+        "stripe==2.42.0",
```
Comment (dmosorast) [inline setup.py]: We should think a bit more about bumping this SDK version for the tap, either putting it in extras_require to make just the tests use this version, or making the tests run on version `2.10.1` to keep them consistent.

File: setup.py [hunk 2]
```diff
@@ -12,12 +12,15 @@
     install_requires=[
         "singer-python==5.5.1",
         "stripe==2.10.1",
+        "pylint==2.1.1",
+        "astroid==2.1.0",
```
Comment (dmosorast) [inline setup.py]: `pylint` and `astroid` should still be dev dependencies.

File: .circleci/config.yml [hunk 3]
```diff
@@ -30,13 +30,15 @@ jobs:
             aws s3 cp s3://com-stitchdata-dev-deployment-assets/environments/tap-tester/sandbox dev_env.sh
             source dev_env.sh
             source /usr/local/share/virtualenvs/tap-tester/bin/activate
-            run-a-test --tap=tap-stripe \
+            pip install 'stripe==2.42.0'
```
Comment (dmosorast) [inline .circleci/config.yml]: This should be `pip install .[test]`, if the dependency is going to be managed in the `extras_require`

Comment (luandy64) [reply .circleci/config.yml]: This line is necessary because `pip install .[test]` errors and fails to install `2.42.0`

Comment (dmosorast) [reply .circleci/config.yml]: Then I think the `test` dependencies should be removed if they're not used.

It's a bit weird in that case that this only exists in the circle file, but I think having a block that's not doing anything is a form of misdirection that doesn't sit well with me.

**Pull request merged**